dist/immutable.js

```diff
@@ -1848,18 +1851,22 @@ function zipWithFactory(keyIter, zipper, iters) {
       var steps;
       if (!isDone) {
         steps = iterators.map(function (i) { return i.next(); });
-        isDone = steps.some(function (s) { return s.done; });
+        if (zipAll) { isDone = steps.every(function (s) { return s.done; }); }
+        else { isDone = steps.some(function (s) { return s.done; }); }
       }
+
       if (isDone) {
         return iteratorDone();
       }
+
       return iteratorValue(
         type,
         iterations++,
-        zipper.apply(null, steps.map(function (s) { return s.value; }))
+        zipper.apply(null, steps.map(function (s) { return s.done ? undefined : s.value; }))
```

Comment:
When `s.done` is true, `s.value` is guaranteed to be `undefined`, so is this change necessary?